File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        store = {}
        for index in range(len(nums)):
            if(nums[index] in store):
                return [store[nums[index]], index]
            
            store[target - nums[index]] = index

        return []

File: Data Structures & Algorithms/duplicate-integer/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums: number[]): boolean {
        const store = new Map()

        for(let i = 0; i < nums.length; i++) {
            if(store.has(nums[i])) {
                return true
            }

            store.set(nums[i], true)
        }

        return false
    }
}
